swarm: fuse tests break on linux with go1.9

#### System information

Geth version: `1.6.6-unstable` (unrelated)
OS & Version: Linux
Commit hash : (if `develop`)

#### Expected behaviour
All tests complete successfully (confirmed to be the case for go 1.8.3 or other go versions)


#### Actual behaviour
tests in `swarm/fuse` package fail.
For every PR for swarm, tests with go1.9 on linux fail, due to the tests hanging and not completing.
For example, see https://travis-ci.org/ethereum/go-ethereum/jobs/280621055

#### Steps to reproduce the behaviour
* Use go1.9 (install if needed first)
* Run `swarm/fuse`package tests: `go test -v github.com/ethereum/go-ethereum/swarm/fuse`

#### Backtrace

````
$ go test -v github.com/ethereum/go-ethereum/swarm/fuse
=== RUN   TestFUSE
=== RUN   TestFUSE/mountListAndUmount
=== RUN   TestFUSE/maxMounts
=== RUN   TestFUSE/remount
=== RUN   TestFUSE/unmount
=== RUN   TestFUSE/unmountWhenResourceBusy
=== RUN   TestFUSE/seekInMultiChunkFile
=== RUN   TestFUSE/createNewFile
=== RUN   TestFUSE/createNewFileInsideDirectory
=== RUN   TestFUSE/createNewFileInsideNewDirectory
=== RUN   TestFUSE/removeExistingFile
=== RUN   TestFUSE/removeExistingFileInsideDir
=== RUN   TestFUSE/removeNewlyAddedFile
=== RUN   TestFUSE/addNewFileAndModifyContents
panic: runtime error: slice bounds out of range [recovered]
	panic: runtime error: slice bounds out of range

goroutine 1548 [running]:
testing.tRunner.func1(0xc42014c000)
	/usr/lib/go/src/testing/testing.go:711 +0x2d2
panic(0x90fd80, 0xed4160)
	/usr/lib/go/src/runtime/panic.go:491 +0x283
internal/poll.(*FD).Write(0xc42092edc0, 0xc420016a50, 0x6, 0x8, 0x0, 0x0, 0x0)
	/usr/lib/go/src/internal/poll/fd_unix.go:218 +0x380
os.(*File).write(0xc420750130, 0xc420016a50, 0x6, 0x8, 0xc420750130, 0x6, 0x0)
	/usr/lib/go/src/os/file_unix.go:233 +0x4e
os.(*File).Write(0xc420750130, 0xc420016a50, 0x6, 0x8, 0x0, 0x0, 0x0)
	/usr/lib/go/src/os/file.go:140 +0x72
github.com/ethereum/go-ethereum/swarm/fuse.(*testAPI).addNewFileAndModifyContents(0xc42000e2f0, 0xc42014c000)
	/home/fabio/go/src/github.com/ethereum/go-ethereum/swarm/fuse/swarmfs_test.go:660 +0xcfd
github.com/ethereum/go-ethereum/swarm/fuse.(*testAPI).(github.com/ethereum/go-ethereum/swarm/fuse.addNewFileAndModifyContents)-fm(0xc42014c000)
	/home/fabio/go/src/github.com/ethereum/go-ethereum/swarm/fuse/swarmfs_test.go:833 +0x34
testing.tRunner(0xc42014c000, 0xc420b66220)
	/usr/lib/go/src/testing/testing.go:746 +0xd0
created by testing.(*T).Run
	/usr/lib/go/src/testing/testing.go:789 +0x2de

````

--- a/swarm/fuse/fuse_file.go
+++ b/swarm/fuse/fuse_file.go
@@ -134,7 +134,7 @@ func (sf *SwarmFile) Write(ctx context.Context, req *fuse.WriteRequest, resp *fu
 		if err != nil {
 			return err
 		}
-		resp.Size = int(sf.fileSize)
+		resp.Size = len(req.Data)
 	} else {
 		log.Warn("Invalid write request size(%v) : off(%v)", sf.fileSize, req.Offset)
 		return errInvalidOffset

